paddle/framework/lod_tensor.h

```diff
@@ -32,7 +34,8 @@ template <typename T>
 using Vector = std::vector<T>;
 #else
 template <typename T>
-using Vector = thrust::host_vector<T>;
+using Vector = thrust::host_vector<
```

Comment:
just `thrust::device_vector` is OK.

Reply:
Yes, I test the `thrust::device_vector`, it also can be used. But how to get the raw pointer is different for `CPU` and `GPU`. 

For example:
```cpp
std::vector<thrust::device_vector<size_t>> lod = {{0, 2 ,4}}

// get the pointer in CPU.
size_t* cpu_ptr = lod[0].data()

// get the pointer for GPU kernel
size_t* gpu_ptr = thrust::raw_pointer_cast(lod[0].data())
```
Both the two methods can satisfy our demands. But for the different usage for CPU and GPU, I'd like to use `thrust::system::cuda::experimental::pinned_allocator<T>`, I can change it to `thrust::device_vector` in the futeaure if there is any question.

Reply:
Why should we need the raw pointer? Maybe just to use `thrust::transform` is OK

Reply:
@reyoung For the sequence operator, we need to use the lod as the input argument of CUDA kernel. The raw pointer (size_t*) is needed. I'm not sure that the `thrust::transform` can be always used for any kernel.